### PR TITLE
feat(tilepagination): support overflow menu in tile pagination

### LINKED ISSUE #1013: [TilePagination] Support overflow menu in tile pagination
Show overflow menu in Tile Pagianation.

Design. 
image

### PR DESCRIPTION
**Pull request by jessieyan**
Closes #1013 

**Summary**
Overflow menu will show up if page number is larger than 7.

image

### CODE REVIEW

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx [hunk 1]
```diff
@@ -86,13 +86,105 @@ const TilePagination = ({ page, numPages, onChange, i18n }) => {
       {pageNumber}
     </button>
   );
+  const getPageSelect = (pageNumber, accumulator) => (
+    <li className="bx--pagination-nav__list-item">
+      <div className="bx--pagination-nav__select">
+        <select
+          className="bx--pagination-nav__page bx--pagination-nav__page--select"
+          data-page-select
+          aria-label="select page number"
```
Comment (stuckless) [inline src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: localization

**Comment by jessieyan** [comment]
> Does this show up in a story, somewhere, so that we can test for RTL and accesiblity?

Yeah, in the base story if you set "number of tile" to a large number. I can also create a separate story for it.

**Comment by jessieyan** [comment]
I found a bug myself. The front overflow menu would have more pages numbers if any of the last 4 page numbers is selected. I'm fixing it.

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx [hunk 2]
```diff
@@ -86,13 +86,105 @@ const TilePagination = ({ page, numPages, onChange, i18n }) => {
...
+              {i + accumulator}
+            </option>
+          ))}
+        </select>
+        <div className="bx--pagination-nav__select-icon-wrapper">
+          <svg
+            focusable="false"
+            preserveAspectRatio="xMidYMid meet"
+            xmlns="http://www.w3.org/2000/svg"
+            className="bx--pagination-nav__select-icon"
```
Comment (sstone2423) [inline src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Needs carbon prefix variable.

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx [hunk 3]
```diff
@@ -86,13 +86,105 @@ const TilePagination = ({ page, numPages, onChange, i18n }) => {
...
+          onChange={evt => onChange(Number(evt.target.value))}
+        >
+          <option value="default" hidden data-page="" />
+          {Array.from({ length: pageNumber }, (v, i) => (
+            <option value={i + accumulator} data-page={i + accumulator}>
+              {i + accumulator}
+            </option>
+          ))}
+        </select>
+        <div className="bx--pagination-nav__select-icon-wrapper">
```
Comment (sstone2423) [inline src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Needs carbon prefix variable.
Is the reasoning to use this specific carbon classname to inherit those styles?

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx [hunk 4]
```diff
@@ -86,13 +86,105 @@ const TilePagination = ({ page, numPages, onChange, i18n }) => {
...
+        <li className="bx--pagination-nav__list-item">{getPageButton(numPages - 2)}</li>,
+        <li className="bx--pagination-nav__list-item">{getPageButton(numPages - 1)}</li>,
+      ];
+    }
+
+    return buttons;
+  };
+
   return (
     <nav className="bx--pagination-nav" ariaLabel={i18n.ariaLabelPagination}>
```
Comment (sstone2423) [inline src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Needs carbon prefix variable alone with the classes below this line

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx (hunk 3 above)
Comment (jessieyan) [reply src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Yes, we are using its own style in carbon. This is carbon pagination but not a component. Carbon only provides guidance. We create our own pagination component base on the guidance.

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx (hunk 1 above)
Comment (jessieyan) [reply src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Done. 
And there is a new story under TileCatalogNew/With OverflowMenu in Pagination.

File: src/components/TileCatalogNew/TilePagination/TilePagination.jsx (hunk 3 above)
Comment (jessieyan) [reply src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: > Needs carbon prefix variable.
> Is the reasoning to use this specific carbon classname to inherit those styles?

Ok, we will still put prefix for these classnames. Working on it.

Comment (sstone2423) [reply src/components/TileCatalogNew/TilePagination/TilePagination.jsx]: Sounds good. Thanks

**Pull request merged**